tools/ktx/command_encode.cpp

```diff
@@ -220,9 +251,18 @@ void CommandEncode::executeEncode() {
 
     MetricsCalculator metrics;
     metrics.saveReferenceImages(texture, options, *this);
-    ret = ktxTexture2_CompressBasisEx(texture, &options);
-    if (ret != KTX_SUCCESS)
-        fatal(rc::IO_FAILURE, "Failed to encode KTX2 file with codec \"{}\". KTX Error: {}", ktxErrorString(ret));
+
+    if (options.vkFormat != VK_FORMAT_UNDEFINED) {
+       options.mode = KTX_PACK_ASTC_ENCODER_MODE_LDR; // TODO: Fix me for HDR textures
+       ret = ktxTexture2_CompressAstcEx(texture, &options);
+       if (ret != KTX_SUCCESS)
+           fatal(rc::IO_FAILURE, "Failed to encode KTX2 file with codec \"{}\". KTX Error: {}", ktxErrorString(ret));
+    } else {
+       ret = ktxTexture2_CompressBasisEx(texture, &options);
+       if (ret != KTX_SUCCESS)
+           fatal(rc::IO_FAILURE, "Failed to encode KTX2 file with codec \"{}\". KTX Error: {}", ktxErrorString(ret));
```

Comment:
The codec type argument is missing.

Reply:
Fixed in https://github.com/KhronosGroup/KTX-Software/pull/952/commits/3645a7aff46fda0206c2bd32f4f38beaec26b26a